src/org/isf/medicals/service/MedicalsIoOperations.java

```diff
@@ -87,39 +86,34 @@ public Medical getMedical(
 	
 	/**
 	 * Retrieves all stored {@link Medical}s.
-	 * If a description value is provides the medicals are filtered.
+	 * If a description value is provided the medicals are filtered.
 	 * @param type the medical type description.
-	 * @nameSorted if <code>true</code> return the list in alphatecial order, by code otherwise
+	 * @param nameSorted if <code>true</code> return the list in alphabetical order, by code otherwise
 	 * @return the stored medicals.
 	 * @throws OHServiceException if an error occurs retrieving the stored medicals.
 	 */
 	public ArrayList<Medical> getMedicals(String type, boolean nameSorted) throws OHServiceException {
-		ArrayList<Medical> medicals = null;
-
-		List<Object> parameters = new ArrayList<Object>();
-
-		StringBuilder query = new StringBuilder();
-		query.append("SELECT * FROM MEDICALDSR JOIN MEDICALDSRTYPE ON MDSR_MDSRT_ID_A = MDSRT_ID_A ");
+		List<Medical> medicals;
 		
 		if (type != null) {
 			
 			if (nameSorted) {
-				medicals = (ArrayList<Medical>)repository.findAllWhereTypeOrderByDescription(type);
+				medicals = repository.findAllWhereTypeOrderByDescription(type);
 			} else {
-				medicals = (ArrayList<Medical>)repository.findAllWhereTypeOrderBySmartCodeAndDescription(type);
+				medicals = repository.findAllWhereTypeOrderBySmartCodeAndDescription(type);
 			}
 			
 		} else {
 
 			if (nameSorted) {
 				medicals = getMedicals(null);
 			} else {
-				medicals = (ArrayList<Medical>)repository.findAllOrderBySmartCodeAndDescription();
+				medicals = repository.findAllOrderBySmartCodeAndDescription();
 			}
 
 		}
 
-		return medicals;
+		return medicals == null ? new ArrayList<Medical>() : new ArrayList<Medical>(medicals);
```

Comment:
is it possible that medicals could be null? jpa repository will return empty list if nothing is found

Reply:
This is not very formally defined in the previous code... so we can follow/introduce the JPA standard keeping in mind that whatever change we do in 'core' must reflect and work as well in 'gui' component and for org.isf.test.Tests

Reply:
I think @tomek39856 you are right. @mwithi I think I don't fully understand what you mean here. The tests are all green (I executed them before each push) and there was nothing changed on the method signature or something. There shouldn't be reason that the tests or clients like the gui might be affected by this change imo. 

I would change this to: return new ArrayList<Medical>(medicals); (as medicals will never be null)

Reply:
@marcoebbinghaus
> There shouldn't be reason that the tests or clients like the gui might be affected by this change imo

It could, because if a check is performed as `resultset == null` instead of `resultset.isEmpty()`, the change would cause the first one to be always `true` and so can lead to unexpected results in the gui if not verified.

Reply:
I still think it could not. I don't understand the isEmpty- and == null- comparison either. There was no check (ifempty or == null) in the former code (before my change) and so it always returned simply the assigned value (which could never be null, because the value always came from a repository method call, which never returns null but an empty list in case of a query that didn't return at least one entity).

 There are four places where medicals is assigned a value: 

- `medicals = repository.findAllWhereTypeOrderByDescription(type);`
- `medicals = repository.findAllWhereTypeOrderBySmartCodeAndDescription(type);`
- `medicals = getMedicals(null);`
- `medicals = repository.findAllOrderBySmartCodeAndDescription();`

the third one will result in:

`medicals = (ArrayList<Medical>)repository.findAllWhereDescriptionOrderByDescription(description);` or `medicals = (ArrayList<Medical>)repository.findAllByOrderByDescription();`

So where ever the value of medicals list might be set, the assigned value will come from a repository which will never return null (empty list in worst case).

Therefore the null check is superfluous - as @tomek39856 said - and I would change it to what I proposed above. Also I would extract both outer cases (the block of `if (type != null)` and the correspondig `else` block. In that method into two private methods as suggested by @tomek39856 and I would clarify in the JavaDoc that those methods will never return null.

Please have a look at my newest push, which should clarify what I mean. And if you don't agree please tell me what I should change to code to.

Reply:
Sorry @marcoebbinghaus, I thought the comparation `medicals == null` was already there for some reason, hence your decision to maintain it. I agree with @tomek39865 that is superfluous and so with your last push.

I was talking as general way to proceed when changing return values between 'core' layers (like between 'service' and 'manager') and/or 'gui'.